contracts/button/lib.rs

```diff
@@ -169,13 +200,13 @@ pub mod button_game {
         /// Deadline is the block number at which the game will end if there are no more participants
         #[ink(message)]
         pub fn deadline(&self) -> BlockNumber {
-            self.last_press + self.button_lifetime
+            self.data.get_or_default().last_press + self.data.get_or_default().button_lifetime
```

Comment:
My preferred choice would be to panic here, rather than returning the default. It seems dangerous to return some arbitrary (default) data to the client who can't tell whether they're some defaults or actual ones. Falling back to the defaults would also mean the data was not properly initialized, which I would consider a setup error.

Reply:
I actually agree with @deuszx here -- self.data.get().unwrap() would be better, even though it looks ugly. Making it default can cover up some issues that we might otherwise be able to find during testing.

Reply:
ok - btw do we know if repeated calls to data.get() are cached?